UnityProject/Assets/Scripts/Items/Weapons/SawnOff.cs

```diff
@@ -58,42 +58,40 @@ public bool WillInteract(InventoryApply interaction, NetworkSide side)
 
 	public void ServerPerformInteraction(InventoryApply interaction)
 	{
-		if (interaction.TargetObject == gameObject && interaction.IsFromHandSlot)
+		if (interaction.TargetObject != gameObject && !interaction.IsFromHandSlot) return;
```

Comment:
shouldn't these first 2 ifs be part of the WillInteract method instead? Since we do nothing when any of them fail.